go.mod

```diff
@@ -3,6 +3,7 @@ module github.com/livekit/livekit-server
 go 1.18
 
 require (
+	github.com/MicahParks/peakdetect v0.0.6
```

Comment:
Using a change point detector/peak detector as described here - https://stackoverflow.com/questions/22583391/peak-signal-detection-in-realtime-timeseries-data/22640362#22640362. The idea is to run this the difference of delta of inter-packet-arrival (i. e. if two packets are sent with a time gap, ideally they should be received with the same time gap, so the difference of those two time gaps should be 0, but when congested, the receive gap will build as queues build up) and get a congestion signal.